core/domain/suggestion_services.py

```diff
@@ -542,3 +556,59 @@ def check_can_resubmit_suggestion(suggestion_id, user_id):
     suggestion = get_suggestion_by_id(suggestion_id)
 
     return suggestion.author_id == user_id
+
+
+def check_can_edit_suggestion(user, suggestion_id):    # pylint: disable=too-many-return-statements
```

Comment:
is this resolved?

Reply:
I think this is a good solution. Instead of checking different type rights its better to make one general activity type and check its rights.
Thanks for telling this I was missing this.

Reply:
done PTAL